Ghidra/Processors/8048/data/languages/8048.slaspec

```diff
@@ -0,0 +1,349 @@
+Cc: "NT1" is ophi=4 { tmp:1 = getT1(); tmp = !tmp; export tmp; }
+Cc: "NZ"  is ophi=9 { tmp:1 = A!=0; export tmp; }
+Cc: "TF"  is ophi=1 { tmp:1 = getTF(); export tmp; }
+Cc: "T0"  is ophi=3 { tmp:1 = getT0(); export tmp; }
+Cc: "T1"  is ophi=5 { tmp:1 = getT1(); export tmp; }
+Cc: "Z"   is ophi=12 { tmp:1 = A==0; export tmp; }
+
+
+# Conventience tables for opcodes taking both Rn and Ri (and Imm)
+Rni: Rn is Rn { export Rn; }
+Rni: Ri is Ri { export Ri; }
+RniI: Rni is Rni { export Rni; }
+RniI: Imm is Imm { export Imm; }
+
+
+# These are not decoded correctly if placed alphabetically...
+:JB^Ab Addr8    is oplo=2 & abfill=1 & Ab; Addr8 { if(Ab) goto Addr8; }
+:J^Cc Addr8     is oplo=6 & Cc; Addr8 { if(Cc) goto Addr8; }
+:CALL Addr12    is aaddrfill=1 & aoplo=4 & Addr12 { funcall(Addr12); }
+:JMP Addr12     is aaddrfill=0 & aoplo=4 & Addr12 { goto Addr12; }
+:DJNZ Rn,Addr8  is ophi=14 & Rn; Addr8 { Rn = Rn - 1; if(Rn != 0) goto Addr8; }
+
+:ADD A,Rni      is ophi=6 & A & Rni { add(A,A,Rni,0); }
+:ADD A,Imm      is (ophi=0 & A)... & Imm { add(A,A,Imm,0); }
+:ADDC A,Rni     is ophi=7 & A & Rni { add(A,A,Rni,C); }
+:ADDC A,Imm     is (ophi=1 & A)... & Imm { add(A,A,Imm,C); }
+:ANL A,RniI     is (ophi=5 & A)... & RniI { A = A & RniI; }
+:ANL Pp,Data    is ophi=9 & Pp; Data { Pp = Pp & Data; }
+:ANLD Xpp,A     is ophi=9 & Xpp & A { Xpp = Xpp & (A & 0xf); }
+:CLR A          is ophi=2 & oplo=7 & A { A = 0; }
+:CLR C          is ophi=9 & oplo=7 & C { C = 0; }
+:CLR F0         is ophi=8 & oplo=5 & F0 { F0 = 0; }
+:CLR F1         is ophi=10 & oplo=5 & F1 { F1 = 0; }
+:CPL A          is ophi=3 & oplo=7 & A { A = ~A; }
+:CPL C          is ophi=10 & oplo=7 & C { C = !C; }
+:CPL F0         is ophi=9 & oplo=5 & F0 { F0 = !F0; }
+:CPL F1         is ophi=11 & oplo=5 & F1 { F1 = !F1; }
+:DA A           is ophi=5 & oplo=7 & A { da(A); }
+:DEC A          is ophi=0 & oplo=7 & A { A = A - 1; }
+:DEC Rn         is ophi=12 & Rn { Rn = Rn - 1; }
+:DIS ExtInt     is ophi=1 & oplo=5 & ExtInt { disableExtInt(); }
+:DIS TCntInt    is ophi=3 & oplo=5 & TCntInt { disableTCntInt(); }
+:EN ExtInt      is ophi=0 & oplo=5 & ExtInt { enableExtInt(); }
+:EN TCntInt     is ophi=2 & oplo=5 & TCntInt { enableTCntInt(); }
+:ENT0 Clk       is ophi=7 & oplo=5 & Clk { enableClockOutput(); }
+:IN A,Pp        is ophi=0 & pp!=0 & A & Pp { A = Pp; }
+:INC A          is ophi=1 & oplo=7 & A { A = A + 1; }
+:INC Rni        is ophi=1 & Rni { Rni = Rni + 1; }
+:INS A,Bus      is ophi=0 & oplo=8 & A & Bus { A = Bus; }
+:JMPP AddrInd   is ophi=11 & oplo=3 & AddrInd { goto AddrInd; }
+:MOV A,Imm      is (ophi=2 & A)... & Imm { A = Imm; }
+:MOV A,Psw      is ophi=12 & oplo=7 & A & Psw { getPSW(A); }
+:MOV A,Rni      is ophi=15 & A & Rni { A = Rni; }
+:MOV A,Tmr      is ophi=4 & oplo=2 & A & Tmr { A = getTmr(); }
+:MOV Psw,A      is ophi=13 & oplo=7 & Psw & A { setPSW(A); }
+:MOV Rni,A      is ophi=10 & Rni & A { Rni = A; }
+:MOV Rni,Data	is ophi=11 & Rni; Data { Rni = Data; }
```

Comment:
The semantics are blend in with the patterns.
I think formatted on a separate line is more readable.

:MOV Rni,Data is ophi=11 & Rni; Data {
        Rni = Data;
}
:MOV Tmr,A    is ophi=6 & oplo=2 & Tmr & A {
        setTmr(A);
}
:MOVD A,Xpp   is ophi=0 & Xpp & A {
        A = (Xpp & 0xf);
}
:MOVD Xpp,A   is ophi=3 & Xpp & A {
        Xpp = (A & 0xf);
}
:MOVP A,PData is ophi=10 & oplo=3 & A & PData {
        A = PData;
}

Reply:
Personally I find it a lot easier to read with one instruction on each line and the three sections in three columns, than with these different types of lines blending in with each other.
But there is no accounting for taste, so I reformatted it for you.  :smirk: